addons/hr_timesheet/models/project_task.py

```diff
@@ -156,14 +156,26 @@ def _get_groups(self):
 
     def action_view_subtask_timesheet(self):
         self.ensure_one()
+        is_internal_user = self.env.user.has_group('base.group_user')
         task_ids = self.with_context(active_test=False)._get_subtask_ids_per_task_id().get(self.id, [])
         action = self.env["ir.actions.actions"]._for_xml_id("hr_timesheet.timesheet_action_all")
         graph_view_id = self.env.ref("hr_timesheet.view_hr_timesheet_line_graph_by_employee").id
         new_views = []
         for view in action['views']:
+            if not is_internal_user:
+                if view[1] == 'tree':
+                    new_views.append((self.env.ref("hr_timesheet.hr_timesheet_line_portal_tree").id, 'tree'))
```

Comment:
To have your new views, the user has to update the module. Since we cannot guarantee he will do it, we should check if the view is there before replacing the views in the action. 😅 

```suggestion
                   tree_view_id = self.env['ir.model.data']._xmlid_to_res_id('hr_timesheet.hr_timesheet_line_portal_tree')
                   if tree_view_id:
                       
new_views.append(tree_view_id, 'tree'))
```

same comment for the other views. 🙂 

I use `self.env['ir.model.data']._xmlid_to_res_id` instead of `ref` because `ref` will do an extra query to check if the record exists and since we just need the id we could directly use `self.env['ir.model.data']._xmlid_to_res_id` to get the id instead of the recordset. 🙂

Reply:
but then, this means that a part of the bug fix won't be effective, unless the user does a module update. This is annoying, but i suppose there is no way around it. Is it why you let the part inside the _compute method to set the field ts_readonly to true ?

Reply:
Yep, we can let the compute as you did. 🙂 

And indeed, we have no other alternatives. 😕